File: js/queue.js
function MyException(message) {
   this.message = message;
   this.name = 'MyException';
}

function Queue(maxLength=0) {
  if (typeof maxLength != 'number' || maxLength < 0) throw new MyException('Incorrect length!');
  this.queue = [];
  this.maxLength = maxLength;

  this.add = function(obj) {
    if (this.queue.length < this.maxLength) {
      this.queue.push(obj);
    } else {
      console.log('Queue is full now');
      //throw new MyException('Queue is full now');
    }
    return this;
  }
  this.remove = function() {
    this.queue.pop(0);
    return this;
  }
  this.clean = function() {
    this.queue.length = 0;
    return this;
  };
  return this;
};


module.exports = {
  Queue: Queue,
  MyException: MyException
}

File: python/queue.py
class Queue:
    def __init__(self, max_length=0):
        if type(max_length) != int or max_length < 0:
            raise Exception
        self.max_length = max_length
        self.queue = []

    def add(self, obj):
        if self.max_length > len(self.queue):
            self.queue.append(obj)
        else:
            print('Queue if full')
        return self

    def remove(self):
        try:
            self.queue.pop(0)
        except IndexError:
            print('Cannot remove element from queue - Queue is empty')
        return self

    def clear(self):
        self.queue = []
        return self
